sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/DirectoryAPITest.groovy

```diff
@@ -230,6 +248,164 @@ class DirectoryAPITest extends APISpec {
+        setup:
+        def metadata = new HashMap<String, String>()
+        if (key1 != null && value1 != null) {
+            metadata.put(key1, value1)
+        }
+        if (key2 != null && value2 != null) {
+            metadata.put(key2, value2)
+        }
+        def options = new DataLakePathCreateOptions().setMetadata(metadata)
+
+        expect:
+        dc.createWithResponse(options, null, null).getStatusCode() == statusCode
+        // Directory adds a directory metadata value
+        for(String k : metadata.keySet()) {
+            dc.getProperties().getMetadata().containsKey(k)
+            dc.getProperties().getMetadata().get(k) == metadata.get(k)
+        }
+
+        where:
+        key1  | value1 | key2   | value2 || statusCode
+        null  | null   | null   | null   || 201
+        "foo" | "bar"  | "fizz" | "buzz" || 201
+    }
+
+    def "Create options with permissions and umask"() {
+        setup:
+        dc = fsc.getDirectoryClient(generatePathName())
+        def permissions = "0777"
+        def umask = "0057"
+        def options = new DataLakePathCreateOptions().setPermissions(permissions).setUmask(umask)
+        dc.createWithResponse(options, null, null)
+
+        when:
+        def acl = dc.getAccessControlWithResponse(true, null, null, null).getValue()
+
+        then:
+        PathPermissions.parseSymbolic("rwx-w----").toString() == acl.getPermissions().toString()
+
+    }
+
+    def "Create options with lease id"() {
+        when:
+        def leaseId = UUID.randomUUID().toString()
+        dc = fsc.getDirectoryClient(generatePathName())
+        def options = new DataLakePathCreateOptions().setProposedLeaseId(leaseId).setLeaseDuration(15)
+        def response = dc.createWithResponse(options, null, null)
+
+        then:
+        // assert lease id not supported for directory
+        thrown(IllegalArgumentException)
+    }
+
+    def "Create options with lease duration"() {
+        when:
+        dc = fsc.getDirectoryClient(generatePathName())
+        def leaseId = UUID.randomUUID().toString()
+        def options = new DataLakePathCreateOptions().setLeaseDuration(15).setProposedLeaseId(leaseId)
+        def response = dc.createWithResponse(options, null, null)
+
+        then:
+        // assert lease duration not supported for directory
+        thrown(IllegalArgumentException)
+    }
+
+    def "Create options with time expires on"() {
+        when:
+        dc = fsc.getDirectoryClient(generatePathName())
+        def leaseId = UUID.randomUUID().toString()
+        def options = new DataLakePathCreateOptions().setExpiresOn(OffsetDateTime.now())
+            .setProposedLeaseId(leaseId).setExpiryOptions(PathExpiryOptions.ABSOLUTE)
+        dc.createWithResponse(options, null, null)
+
+        then:
+        // assert expires on not supported for directory
+        thrown(IllegalArgumentException)
+
+    }
+
+    def "Create options with time to expire"() {
```

Comment:
These four tests that are checking for IllegalArgumentExceptions can probably be put into one test with a data table